src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1684,6 +1822,132 @@ UndistributeTable(Oid relationId)
 	{
 		ereport(ERROR, (errmsg("could not finish SPI connection")));
 	}
+
+	if (cascadeToColocated)
+	{
+		Oid colocatedTableId = InvalidOid;
+
+		/* For now we only support cascade to colocation for alter_distributed_table UDF */
+		Assert(conversionType == ALTER_DISTRIBUTED_TABLE);
+		foreach_oid(colocatedTableId, colocatedTableList)
+		{
+			if (colocatedTableId == relationId)
+			{
+				continue;
+			}
+			if (conversionType == ALTER_DISTRIBUTED_TABLE)
+			{
+				AlterDistributedTable(colocatedTableId, NULL, shardCount, relationName,
```

Comment:
be careful that colocateWithTableName in CreateDistributedTable is expected to have the schema name as well, whereas relationName excludes the schema name.